### PR TITLE
feat(besu): add prometheus exporter

### LINKED ISSUE #533: feat(besu): add prometheus metrics support to the OpenAPI specs of the plugin
### Description
As a developer, I want to integrate Prometheus Exporter for the Hyperledger Besu Cactus plugin, Issue #458 details the description in depth and covers up for all the underlying plugins.

cc: @petermetz @takeutak @sfuji822 @hartm @jonathan-m-hamilton @AzaharaC @jordigiam @kikoncuo

### PR DESCRIPTION
**Pull request by jagpreetsinghsasan**
## Commit to be reviewed
------------------------------

feat(besu): prometheus exporter metrics integration

	Primary Change
	--------------

	1. The besu ledger connector plugin now includes the prometheus metrics exporter integration
	2. OpenAPI spec now has api endpoint for the getting the prometheus metrics

	Refactorings that were also necessary to accomodate 1) and 2)
	------------------------------------------------------------

	3. GetPrometheusMetricsV1 class is created to handle the corresponding api endpoint
	4. IPluginLedgerConnectorBesuOptions interface in PluginLedgerConnectorBesu class now has a prometheusExporter optional field
	5. The PluginLedgerConnectorBesu class has relevant functions and codes to incorporate prometheus exporter
	6. Added Readme.md on the prometheus exporter usage

	Fixes #533

Signed-off-by: Jagpreet Singh Sasan <jagpreet.singh.sasan@accenture.com>

### CODE REVIEW

File: packages/cactus-plugin-ledger-connector-besu/src/main/json/openapi.json [hunk 1]
```diff
@@ -552,6 +552,35 @@
...
+                    }
+                }
+            },
+            "PrometheusExporterMetricsRequest": {
+                "type": "object",
+                "required": [
+                    "promExporter"
+                ],
+                "properties": {
+                    "promExporter":{
```
Comment (petermetz) [inline packages/cactus-plugin-ledger-connector-besu/src/main/json/openapi.json:575]: @jagpreetsinghsasan Is the `promExporter` being used?

Comment (jagpreetsinghsasan) [reply packages/cactus-plugin-ledger-connector-besu/src/main/json/openapi.json]: Fixed.

**Review by takeutak — APPROVED** [review]

**Comment by petermetz** [comment]
@jagpreetsinghsasan The GH Actions CI seems to be stuck on this one as well so you have a couple of options to try and kick it back into action
1. Push a dummy commit (that later you can and should squash back into the original one) and see if this re-triggers the CI run
2. If 1) fails then close this PR and open a brand new one with the exact same content
3. If 2) also fails then it's time to contact Github support, but I'm hoping 1 or 2 will fix it.

**Review by takeutak — APPROVED** [review]

File: packages/cactus-plugin-ledger-connector-besu/README.md [hunk 2]
```diff
@@ -33,7 +34,7 @@ In the project root folder, run this command to compile the plugin and create th
 npm run tsc
 ```
 
-## Usage
+This class creates a prometheus exporter, which scraps the transactions (total transaction count) for the use cases incorporating the use of Besu connector plugin.
```
Comment (petermetz) [inline packages/cactus-plugin-ledger-connector-besu/README.md:37]: @jagpreetsinghsasan Typo? `scraps` -> `scrapes`

**Review by petermetz — CHANGES_REQUESTED** [review]

File: packages/cactus-plugin-ledger-connector-besu/README.md [hunk 3]
```diff
@@ -80,6 +81,45 @@ enum Web3SigningCredentialType {
 ```
 > Extensive documentation and examples in the [readthedocs](https://readthedocs.org/projects/hyperledger-cactus/) (WIP) 
 
+## Prometheus Exporter
+
+This class creates a prometheus exporter, which scraps the transactions (total transaction count) for the use cases incorporating the use of Besu connector plugin.
```
Comment (petermetz) [inline packages/cactus-plugin-ledger-connector-besu/README.md:86]: @jagpreetsinghsasan  Is this paragraph a duplicate of the one I made the typo comment on above?

File: packages/cactus-plugin-ledger-connector-besu/README.md [hunk 4]
```diff
@@ -80,6 +81,45 @@ enum Web3SigningCredentialType {
...
+
+### Usage
+The prometheus exporter object is initialized in the `PluginLedgerConnectorBesu` class constructor itself, so instantiating the object of the `PluginLedgerConnectorBesu` class, gives access to the exporter object.
+You can also initialize the prometheus exporter object seperately and then pass it to the `IPluginLedgerConnectorBesuOptions` interface for `PluginLedgerConnectoBesu` constructor.
+
+`getPrometheusExporterMetricsEndpointV1` function returns the prometheus exporter metrics, currently displaying the total transaction count, which currently increments everytime the `transact()` method of the `PluginLedgerConnectorBesu` class is called.
```
Comment (petermetz) [inline packages/cactus-plugin-ledger-connector-besu/README.md:92]: @jagpreetsinghsasan My usual thing about the function name not containing `endpoint` :-)

File: packages/cactus-test-tooling/src/main/typescript/besu/besu-test-ledger.ts [hunk 5]
```diff
@@ -18,7 +18,7 @@ export interface IBesuTestLedgerConstructorOptions {
 }
 
 export const BESU_TEST_LEDGER_DEFAULT_OPTIONS = Object.freeze({
-  containerImageVersion: "2021-01-08-7a055c3",
+  containerImageVersion: "latest",
```
Comment (petermetz) [inline packages/cactus-test-tooling/src/main/typescript/besu/besu-test-ledger.ts:21]: @jagpreetsinghsasan Please specify a tag that's designed to not change over time: This is to ensure that the CI/tests consistently succeed or fail while being executed with a given revision (git sha) of the code.

File: packages/cactus-test-tooling/src/main/typescript/fabric/fabric-test-ledger-v1.ts [hunk 6]
```diff
@@ -40,7 +40,7 @@ export interface IFabricTestLedgerV1ConstructorOptions {
  * Provides default options for Fabric container
  */
 const DEFAULT_OPTS = Object.freeze({
-  imageVersion: "2021-01-05-3400c06",
+  imageVersion: "latest",
```
Comment (petermetz) [inline packages/cactus-test-tooling/src/main/typescript/fabric/fabric-test-ledger-v1.ts:43]: @jagpreetsinghsasan Please specify a tag that's designed to not change over time: This is to ensure that the CI/tests consistently succeed or fail while being executed with a given revision (git sha) of the code.

File: packages/cactus-test-tooling/src/main/typescript/quorum/quorum-test-ledger.ts [hunk 7]
```diff
@@ -18,7 +18,7 @@ export interface IQuorumTestLedgerConstructorOptions {
 }
 
 export const QUORUM_TEST_LEDGER_DEFAULT_OPTIONS = Object.freeze({
-  containerImageVersion: "2021-01-08-7a055c3",
+  containerImageVersion: "latest",
```
Comment (petermetz) [inline packages/cactus-test-tooling/src/main/typescript/quorum/quorum-test-ledger.ts:21]: @jagpreetsinghsasan Please specify a tag that's designed to not change over time: This is to ensure that the CI/tests consistently succeed or fail while being executed with a given revision (git sha) of the code.

**Review by petermetz — CHANGES_REQUESTED** [review]

File: packages/cactus-test-tooling/src/main/typescript/besu/besu-test-ledger.ts (hunk 5 above)
Comment (jagpreetsinghsasan) [reply packages/cactus-test-tooling/src/main/typescript/besu/besu-test-ledger.ts:21]: I didnt change it.. Somehow it got changed. Lemme revert these

**Review by petermetz — APPROVED** [review]

**Review by takeutak — DISMISSED** [review]

**Review by takeutak — APPROVED** [review]

**Review by petermetz — APPROVED** [review]

**Comment by petermetz** [comment]
@jagpreetsinghsasan Please rebase this and then it's ready to go! :-)

**Pull request merged**